code/modules/lighting/lighting_atom.dm

```diff
@@ -9,6 +9,8 @@
 	//  Useful for objects like light fixtures that aren't visually in the middle of the turf, but aren't offset either.
 	var/light_offset_x
 	var/light_offset_y
+	/// An override for cases where the light is not facing the same direction as the object.
+	var/light_dir
```

Comment:
(yes, this is a new atom var, but it has no memory cost unless changed, and most things won't have it changed)